src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
 export interface Figure {
+  shape: string,
+  color: string,
+  getArea(): number
 
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: string = 'triangle';
 
-}
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const max = Math.max(this.a, this.b, this.c);
+
+    if (max >= this.a + this.b + this.c - max) {
+      throw new Error('Sides 1, 2 and 3 can\'t form a triangle');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: string;
+
+  constructor(public color: string, public radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+
+    this.color = color;
+    this.shape = 'circle';
+    this.radius = radius;
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: string;
+
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Length of sides must be greater than 0');
+    }
+
+    this.color = color;
+    this.height = height;
```

Comment:
Remove this line as the height is already assigned in the constructor parameters